### PR TITLE
Import TIMIT dataset

### PR DESCRIPTION
**Pull request by mlrobsmt**
Hi,

I created a TIMIT preparation script for everything (excluding the download, as it's LDC). It was one of the few, very well used, paid LDC datasets that I had at my disposal. It's only small, with around 5 hours worth of audio and has 630 speakers with 10 types of specially crafted sentences, designed to test ASR systems. 

"The TIMIT corpus of read speech is designed to provide speech data for acoustic-phonetic studies and for the development and evaluation of automatic speech recognition systems. TIMIT contains broadband recordings of 630 speakers of eight major dialects of American English, each reading ten phonetically rich sentences. The TIMIT corpus includes time-aligned orthographic, phonetic and word transcriptions as well as a 16-bit, 16kHz speech waveform file for each utterance."

More details, such as sentence distribution/gender, can be found [here](https://catalog.ldc.upenn.edu/ldc93s1). Traditionally papers have compared PER using TIMIT but there's no reason why WER cannot be compared also.

**What's included in this PR?**
1. import_timit.py - extracts, processes and outputs the required train/test CSV files 
2. run-timit.sh - will run TIMIT with basic defaults (similarly to the other run scripts)

**Has it been tested?**
Yes - I have tested with Ubuntu 16.04 using python 2.7. It extracts, preprocesses builds the CSVs and runs as expected.

### CODE REVIEW

**Comment by kdavis-mozilla** [comment]
We haven't purchased this data set. So, the review is somewhat theoretical

File: bin/run-timit.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+if [ -d "${COMPUTE_KEEP_DIR}" ]; then
+    checkpoint_dir=$COMPUTE_KEEP_DIR
+else
+    checkpoint_dir=$(python -c 'from xdg import BaseDirectory as xdg; print(xdg.save_data_path("deepspeech/timit"))')
+fi
+
+python -u DeepSpeech.py \
+  --train_files data/TIMIT/timit_train.csv \
+  --dev_files data/TIMIT/timit_test.csv \
```
Comment (kdavis-mozilla) [inline bin/run-timit.sh]: I'd guess this is the "right thing to do" as there's no validation data set in TIMIT.

File: bin/run-timit.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  --test_files data/TIMIT/timit_test.csv \
+  --train_batch_size 8 \
+  --dev_batch_size 8 \
+  --test_batch_size 8 \
+  --n_hidden 494 \
+  --epoch 50 \
+  --checkpoint_dir "$checkpoint_dir" \
+  "$@"
```
Comment (kdavis-mozilla) [inline bin/run-timit.sh]: Do the default values for all other parameters work?

File: bin/import_timit.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    print("Preprocessing Complete")
+    print("Building CSVs")
+
+    # Lists to build CSV files
+    train_list_wavs, train_list_trans, train_list_size = [], [], []
+    test_list_wavs, test_list_trans, test_list_size = [], [], []
+
+    for root, dirnames, filenames in os.walk(target):
+        for filename in fnmatch.filter(filenames, "*_rif.wav"):
+
```
Comment (kdavis-mozilla) [inline bin/import_timit.py]: Could you remove the blank line here

**Comment by jackHuanghello** [comment]
Have you tested the TIMIT data set? The wer value of my result is 26%. Does my result the same to yours?

**Comment by kdavis-mozilla** [comment]
@jackHuanghello Could you please move such conversations to the [Discourse Forum](https://discourse.mozilla.org/c/deep-speech) please.

**Pull request merged**